hackathon_site/dashboard/frontend/src/pages/TeamDetail/TeamDetail.test.tsx

```diff
@@ -1,21 +1,60 @@
 import React from "react";
 
-import { render, screen } from "testing/utils";
+import { render, screen, waitFor } from "testing/utils";
 
-import { mockTeamMultiple } from "testing/mockData";
+import { mockTeam, mockTeamMultiple } from "testing/mockData";
 import TeamDetail, { PageParams } from "pages/TeamDetail/TeamDetail";
 import { RouteComponentProps } from "react-router-dom";
+import { get } from "api/api";
+import { AxiosResponse } from "axios";
 
-test("renders without crashing", () => {
-    const teamDetailProps = {
-        match: {
-            params: {
-                id: mockTeamMultiple.id.toString(),
-            },
+jest.mock("api/api", () => ({
+    ...jest.requireActual("api/api"),
+    get: jest.fn(),
+}));
+
+const mockedGet = get as jest.MockedFunction<typeof get>;
+
+const teamDetailProps = {
+    match: {
+        params: {
+            id: mockTeamMultiple.id.toString(),
         },
-    } as RouteComponentProps<PageParams>;
-    render(<TeamDetail {...teamDetailProps} />);
-    expect(
-        screen.getByText(`Team ${teamDetailProps.match.params.id} Overview`)
-    ).toBeInTheDocument();
+    },
+} as RouteComponentProps<PageParams>;
+
+describe("<TeamDetail />", () => {
+    test("renders loading component and then data without crashing", () => {
+        render(<TeamDetail {...teamDetailProps} />);
+
+        expect(screen.getByTestId("team-info-linear-progress")).toBeInTheDocument();
+
+        expect(
+            screen.getByText(`Team ${teamDetailProps.match.params.id} Overview`)
+        ).toBeInTheDocument();
+
+        expect(mockedGet).toHaveBeenCalledWith(
+            `/api/event/teams/${mockTeamMultiple.id}/`
+        );
+    });
+
+    test("displays 404 error when the requested team id is not found", async () => {
+        const failureResponse = {
+            response: {
+                status: 404,
+                statusText: "Not Found",
+                message: "Could not find team code: Error 404",
+            },
+        };
+
+        mockedGet.mockRejectedValueOnce(failureResponse);
```

Comment:
I didn't catch this earlier but here you are mocking the entire get function. In the future we will need to use the get function for other things so this might cause some problems.
Instead you should use something like:

`when(mockedGet).calledWith(..).mockRejectedValue(failureResponse)`

^^ This might not be completely accurate but take a look at some other unit tests in the repo for some inspiration.

Reply:
i used exactly what you suggested and the test still passes, so i think you're right!